Core/include/Acts/Propagator/MaterialInteractor.hpp

```diff
@@ -119,7 +119,7 @@ struct MaterialInteractor {
       updateResult(state, stepper, result);
     }
 
-    const TrackingVolume* volume = navigator.currentVolume(state.navigation);
+    auto volume = navigator.currentVolume(state.navigation);
```

Comment:
```suggestion
    const auto* volume = navigator.currentVolume(state.navigation);
```
to express this is a pointer to const

Reply:
Could we at least express this in a basic concept that requires the type to be either `TrackingVolume` or `DetectorVolume`?

These two should ideally merge (`DetectorVolume` -> `TrackingVolume`) as soon as possible, @asalzburger.

Reply:
either we get splitting cases like this in main or we setup a feature branch for the new geometry. not sure what is best but I am also not a big fan of starting to specialize on the geometry